commands/utils/docker-endpoint.ts

```diff
@@ -9,8 +10,35 @@ class DockerClient {
     private endPoint: Docker;
 
     constructor() {
-        // Pass no options so that the defaultOpts of docker-modem will be used
-        this.endPoint = new Docker();
+        this.refreshEndpoint();
+    }
+
+    public refreshEndpoint() {
+        const errorMessage = 'The docker.host configuration setting must be entered as <host>:<port>, e.g. dockerhost:2375';
+        const value: string = vscode.workspace.getConfiguration("docker").get("host", "");
+        if (value) {
+            let newHost: string = '';
+            let newPort: number = 2375;
+            let sep: number = -1;
+
+            sep = value.lastIndexOf(':');
+
+            if (sep < 0) {
+                vscode.window.showErrorMessage(errorMessage);
+            } else {
+                newHost = value.slice(0, sep);
+                newPort = Number(value.slice(sep + 1));
+                if (isNaN(newPort)) {
+                    vscode.window.showErrorMessage(errorMessage);
+                } else {
+                    this.endPoint = new Docker({ host: newHost, port: newPort});
+                }
+            }
+        }
+        if (!this.endPoint) {
```

Comment:
We missed a case - when changing from a non-empty endpoint to empty endpoint we need to create a new Endpoint:

Please change 
   if (!this.endPoint) { 
to
   if (!this.endPoint || !value) {